rero_ils/modules/loans/api.py

```diff
@@ -775,6 +807,29 @@ def is_overdue_loan(loan):
     return now > due_date + timedelta(days=days_after)
 
 
+def loan_has_open_events(loan_pid=None):
+    """Check if a loan has open patron transactions.
+
+    Loan has no open_events if the he has no related patron transaction with
+    the status open.
+
+    :return True|False.
+    """
+    notifications = []
+    search = NotificationsSearch().filter(
+        'term', loan__pid=loan_pid).source().scan()
+    for record in search:
+        notifications.append(record.pid)
+    if notifications:
+        for notification in notifications:
+            transactions_count = PatronTransactionsSearch().filter(
+                'term', notification__pid=notification
+                ).filter('term', status='open').source().count()
+            if transactions_count:
+                return True
```

Comment:
```
for record in search:
  transactions_count = PatronTransactionsSearch()
      .filter('term', notification__pid=record.pid)
      .filter('term', status='open')
      .source().count()
  if transction_count:
      return True
```
Using this method, you avoid to loop all values from the first loop + an useless `if` statement